lib/uvm/bridge.browser.js

```diff
@@ -1,142 +1,113 @@
 
         // function to forward messages emitted
         forwardEmits = function (e) {
-            if (!(e && e.data && _.isString(e.data.__emit_uvm) && (e.data.__id_uvm === id))) { return; }
+            if (!(e && e.data && (typeof e.data.__emit_uvm === 'string') && (e.data.__id_uvm === id))) { return; }
 
             var args;
             try { args = Flatted.parse(e.data.__emit_uvm); }
             catch (err) { return bridge.emit(ERROR, err); }
             bridge.emit.apply(bridge, args); // eslint-disable-line prefer-spread
         },
 
-        processCallback = function () {
-            !options._sandbox && iframe.removeEventListener(LOAD, processCallback);
-
-            bridge._dispatch = function () {
-                if (!iframe) {
-                    return bridge.emit(ERROR,
-                        new Error('uvm: unable to dispatch "' + arguments[0] + '" post disconnection.'));
-                }
-
-                iframe.contentWindow.postMessage({
-                    __emit_uvm: Flatted.stringify(Array.prototype.slice.call(arguments)),
-                    __id_uvm: id
-                }, TARGET_ALL);
-            };
-
-            callback(null, bridge);
-        };
-
-    // add event listener for receiving events from iframe (is removed on disconnect)
-    window.addEventListener(MESSAGE, forwardEmits);
-
-    // equip bridge to disconnect (i.e. delete the iframe)
-    bridge._disconnect = function () {
-        if (!iframe) { return; }
-
-        // remove the message handler for this sandbox
-        window.removeEventListener(MESSAGE, forwardEmits);
+        // function to forward errors emitted
+        forwardErrors = function (e) {
+            bridge.emit(ERROR, e);
+        },
 
-        // do not delete sandbox element if not created for the bridge
-        !options._sandbox && iframe.parentNode && iframe.parentNode.removeChild(iframe);
-        iframe = null;
-    };
+        firmwareObjectURL,
+        worker;
 
-    // if sandbox element is provided, we simply need to init with firmware code and wait for load event to be fired
+    // if sandbox worker is provided, we simply need to init with firmware code
+    // @todo validate sandbox type or APIs
     if (options._sandbox) {
-        bridge.once('load.' + id, processCallback); // on load attach the dispatcher
-
-        iframe.contentWindow.postMessage({
-            __init_uvm: sandboxCode(code, id, true) // the last `true` param indicates firmwareOnly
-        }, TARGET_ALL);
-
-        return;
+        worker = options._sandbox;
+        worker.postMessage({ __init_uvm: firmwareCode });
     }
+    // else, spawn a new worker
+    else {
+        // convert the firmware code into a blob URL
+        firmwareObjectURL = window.URL.createObjectURL(
+            new Blob([firmwareCode], { type: 'text/javascript' })
+        );
+
+        worker = new Worker(firmwareObjectURL);
```

Comment:
I think at the beginning of initialising the browser client, we should probe `Blob`, `Window.URL` and `Worker` and raise recognisable errors.